pkg/scaler/prober.go

```diff
@@ -156,8 +158,16 @@ func (p *prober) tryAndRun(prepareRun func() (stopCh <-chan struct{})) error {
 	internalClient, internalSHA, internalErr := p.getClientFromSecret(p.probeDeps.Probe.Internal.KubeconfigSecretName, p.internalSHA)
 	externalClient, externalSHA, externalErr := p.getClientFromSecret(p.probeDeps.Probe.External.KubeconfigSecretName, p.externalSHA)
 
-	// Run a fresh prober goroutine if any one of the secrets changed
-	if internalErr == nil || externalErr == nil {
+	if apierrors.IsNotFound(internalErr) || apierrors.IsNotFound(externalErr) {
+		cancelFn()
+
+		if internalErr != nil {
+			return internalErr
+		}
+
+		return externalErr
+	} else if internalErr == nil || externalErr == nil {
```

Comment:
Then this could be a simple `else`.

```suggestion
	} else {
```

Reply:
then would it result in restart of the probe even if both the secrets "already exist"?